modules/monitoring-resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * monitoring/troubleshooting-monitoring-issues.adoc
+// * support/troubleshooting/investigating-monitoring-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus_{context}"]
+= Resolving the KubePersistentVolumeFillingUp alert firing for Prometheus
+
+As a cluster administrator, you can resolve the `KubePersistentVolumeFillingUp` alert being triggered for Prometheus. 
+
+The critical alert fires when a persistent volume (PV) claimed by a `prometheus-k8s-*` pod in the `openshift-monitoring` project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.
+
+[NOTE]
+====
+There are two `KubePersistentVolumeFillingUp` alerts:
+
+* *Critical alert*: It is triggered when the mounted PV has less than 3% of its total space remaining. 
+* *Warning alert*: It is triggered when the mounted PV has less than 15% of its total space remaining and is expected to fill up within four days.
+====
+
+You can resolve this issue by freeing some space for the PV by removing the three oldest Prometheus time series database (TSDB) blocks.
+
+.Prerequisites
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `cluster-admin` cluster role.
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `dedicated-admin` role.
+endif::openshift-dedicated,openshift-rosa[]
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Remove the three oldest Prometheus TSDB blocks to make more space available for the PV:
```

Comment:
Hello @simonpasquier could you please provide more guidance, what do you think is the best solution here? Do you think it would be enough to maybe put a **Warning** note about the possible dangers? Or do you think it is too risky to include this command at all? How should we therefore deal with this user case? Do we possibly have other methods how to deal with the alert issue?

Thank you in advance for your response!

Reply:
For instance, maybe the users want to copy the data before deleting?

Reply:
@simonpasquier I see so we can provide the instructions to identify which directories could be removed as you suggested (in that case I would ask if you could provide me with these instructions) 

We can also add a warning with the suggestion to the user to backup their Prometheus data before running the command (do we include any instructions how to do that?)

Let me know what you think, thank you!

Reply:
Passing this command to the debug pod returns the size of all blocks, sorted from oldest to newest:
 
````
cd /prometheus/;du -hs $(ls -dt */ | grep -Eo "[0-9|A-Z]{26}")
````

Then users should be able to make a decision on which and how many blocks they need to delete.

Re backup, we don't have a strong history currently so maybe we should just omit this part.

Reply:
Hello @simonpasquier thank you for the command, when I tried to run it, I encountered the following error messages:
```
$ oc debug prometheus-k8s-0 -n openshift-monitoring --image=$(oc get po -n openshift-monitoring 
prometheus-k8s-0 -o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}') -- 
cd /prometheus/;du -hs $(ls -dt */ | grep -Eo "[0-9|A-Z]{26}")

Starting pod/prometheus-k8s-0-debug-bsk5z ...

Removing debug pod ...
du: cannot read directory './.local/share/containers/storage/overlay/
ed1cc9773f4af7028c34ddf6bdd4f2c33b5ad56f5d07c229bb68b3f9f5805cc0': Permission denied
du: cannot read directory './.local/share/containers/storage/overlay/
ef9a0b462fba48388c3920a19b8572958ecc15d781618022e3a9d8f751b5b8ac': Permission denied
du: cannot read directory './.local/share/containers/storage/overlay-containers/
e843b9252161371d561ad012319a6a7e5344fc296a2aa71621c4ff029fe2b7a7/userdata': Permission denied
27G	         .
```
But when I run it like this:
```
$ oc debug prometheus-k8s-0 -n openshift-monitoring --image=$(oc get po -n openshift-monitoring 
prometheus-k8s-0 -o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}')

Starting pod/prometheus-k8s-0-debug-7xj2w ...

sh-5.1$ cd /prometheus/;du -hs $(ls -dt */ | grep -Eo "[0-9|A-Z]{26}")
506M	         .
```
It ran without any of the errors. 

Do you know why the first method did not work as expected?
Thank you!

Reply:
Oh yes, that was the issue, thank you!